docs/getting-started.md

```diff
@@ -33,23 +21,288 @@ If you had installed esy previously:
 npm uninstall --global --update esy
 ```
 
-## Clone & initialize the project
+## Step-by-step tutorial
+
+Let's start with an empty `package.json`
+
+```json
+{}
+```
+
+To write a basic hello world program in OCaml, let's add the `ocaml`
+compiler package to the `package.json`. 
+
+```json
+{
+  "dependencies": {
+    "ocaml": "4.10.x"
+  }
+}
+```
+
+And our OCaml source file, `hello.ml` with 
+
+```ml
+let () = print_endline "Hello, World"
+```
+
+To install the dependencies, run `esy`
+
+```sh
+$ esy
+```
+
+This will fetch `ocaml` and install it in a sandbox exclusively
+created for this newly created project (described by `package.json`)
+
+<screenshot>
+
+Just like in the Yarn/NPM world, build commands are run with a prefix.
+
+To compile,
+
+```sh
+esy ocamlopt -o hello.exe hello.ml
+```
+
+To run the binary, `hello.exe`,
+
+```sh
+./hello.exe
+```
```

Comment:
### Suggested position

5

### Suggested change

If everything worked correctly, running the command above should have output a file called `hello.exe`.
To now execute our program and output _"Hello, World"_, we just need to run:

```sh
./hello.exe
```

If you are following along on a Mac or Linux, it might seem a bit odd, that you can run an `.exe` file, as these files normally can be just run on Windows systems.
In OCaml `.exe` just stands for "executable" and can be run on every system, it was compiled on. So if you compile your program on a Linux, you can run it on other Linux systems, but not on a Mac. To be able to also run your program on a Mac, you would also need to compile it on a Mac.

Reply:
+1